src/win.rs

```diff
@@ -161,6 +163,25 @@ fn test_conout() {
     assert!(conout().is_ok())
 }
 
+#[rustversion::before(1.36)]
+unsafe fn get_console_screen_buffer_info(handle: HANDLE) -> io::Result<CONSOLE_SCREEN_BUFFER_INFO> {
+    let mut buffer_info = ::std::mem::uninitialized();
+    if GetConsoleScreenBufferInfo(handle, &mut buffer_info) == 0 {
+        Err(io::Error::last_os_error())
+    } else {
+        Ok(buffer_info)
+    }
+}
+#[rustversion::since(1.36)]
+unsafe fn get_console_screen_buffer_info(handle: HANDLE) -> io::Result<CONSOLE_SCREEN_BUFFER_INFO> {
+    let mut buffer_info = ::std::mem::MaybeUninit::uninit().assume_init();
```

Comment:
This needs to call `assume_init` _after_ filling the buffer. That is, it should call `GetConsoleScreenBufferInfo(handle, buffer_info.as_mut_ptr()` then `buffer_info.assume_init()` (if there was no error).

We can't take a reference to uninitialized memory (which is why this type was introduced).